packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -770,7 +771,13 @@ class _ViewContentState extends State<_ViewContent> {
         _viewRect = Offset.zero & _screenSize!;
       }
     }
-    unawaited(updateSuggestions());
+
+    if (searchValue != _controller.text) {
+      unawaited(Future<void>(() async {
```

Comment:
IMO this should use a Timer. And the pending timer should be disposed if `State.dispose` is called while the timer is open

Reply:
`unawaited` was there so i can try work around. I applied your suggestion to use `Timer` and it does pass `flutter_hooks` test locally.

Can you please explain why Timer is better? `unawaited` is fairly new.

Reply:
Future internally creates a Timer. But that Timer is not disposed until the callback executes.

So this creates a race condition between the test assertion about "non-disposed Timers" and the callback execution.

If the Timer assert runs before the callback is executed, Future's Timer has yet to be cleaned-up, making the Timer assert fail.

Reply:
I appreciate your suggestion and explanation, it helped a lot.